composition-js/src/validate.ts

```diff
@@ -257,6 +277,19 @@ function initialSubgraphPaths(kind: SchemaRootKind, subgraphs: QueryGraph): Root
   return subgraphs.outEdges(root).map(e => initialState.add(subgraphEnteringTransition, e, noConditionsResolution));
 }
 
+function possibleRuntimeTypeNamesSorted(path: RootPath<Transition>): string[] {
+  const types = path.tailPossibleRuntimeTypes().map((o) => o.name);
+  types.sort((a, b) => a.localeCompare(b));
+  return types;
+}
+
+export function extractValidationError(error: any): ValidationError | undefined {
```

Comment:
Remark: this method is not used within federation itself, but it's useful in some tooling (planning to use it in the update to https://github.com/apollographql/composer-tool after this patch) so I'd prefer keeping it.